src/hip/handlehip.cpp

```diff
@@ -367,23 +367,26 @@ Program Handle::LoadProgram(const std::string& program_name,
                             const std::string& kernel_src) const
 {
     this->impl->set_ctx();
-    params += " -mcpu=" + this->GetDeviceName();
-    auto hsaco = miopen::LoadBinary(
-        this->GetDeviceName(), this->GetMaxComputeUnits(), program_name, params, is_kernel_str);
+    const auto device = this->GetDeviceName();
+    const auto num_cu = this->GetMaxComputeUnits();
+    params += " -mcpu=" + device;
+    auto hsaco = miopen::LoadBinary(device, num_cu, program_name, params, is_kernel_str);
     if(hsaco.empty())
     {
+#if MIOPEN_ENABLE_SQLITE_KERN_CACHE
+        miopen::MarkBinary(device, num_cu, program_name, params, is_kernel_str);
```

Comment:
Am I correct that: If process A calls `MarkBinary` before process B, does `MarkBinary` block process B until process A creates a binary and stores it in the cache?

Reply:
That is correct